src/main/groovy/org/pih/warehouse/inventory/CycleCountUpdateItemCommand.groovy

```diff
@@ -10,6 +14,24 @@ class CycleCountUpdateItemCommand implements Validateable {
 
     boolean recount
 
+    @BindUsing({ obj, source ->
+        def productId = source['inventoryItem']['product'] instanceof Map ? source['inventoryItem']['product']['id'] : source['inventoryItem']['product']
+        def lotNumber = source['inventoryItem']['lotNumber']
+        def expirationDate = source['inventoryItem']['expirationDate'] ? DateUtil.asDate(source['inventoryItem']['expirationDate'].toString()) : null
+
+        Product product = Product.read(productId)
+        InventoryItem inventoryItem = InventoryItem.findByProductAndLotNumber(product, lotNumber) ?: new InventoryItem(product: product, lotNumber: lotNumber)
+
+        if (inventoryItem.expirationDate != expirationDate) {
+            inventoryItem.expirationDate = expirationDate
+            inventoryItem.save()
+        }
+
+        inventoryItem
```

Comment:
this is definetely not a place where updating the date should take place.
Binder should only bind the `InventoryItem` and updating the date should take place in the service.

Reply:
Agreed. Also, I think we'd need to discuss with Manon what the desired behaviour is in this scenario. For example:
- you have a row with lot ABC and expiry Jan 1st 2026
- you add a custom row with lot ABC and expiry Feb 2nd 2027

In this scenario, it's unclear if we should actually update the expiry of lot ABC or if the custom row should just take the expiry from the non-custom row (ie use Jan 1st 2026). Updating the expiry and clicking save draft will change the expiry immediately so would impact anywhere else that uses that item.

and another scenario:
- you have a custom row with lot ABC and expiry Jan 1st 2026
- you edit the custom row to have expiry Feb 2nd 2027

in this scenario it might make sense to have the expiry be updated since the item isn't used anywhere else yet.

Reply:
actually, the first scenario will trigger validation errors on the frontend so shouldn't be possible. You can't have a lot in multiple rows with two different expiry dates.
![image](https://github.com/user-attachments/assets/43087f6a-76c6-4c27-bd87-6481dd8c3821)

I asked Manon on slack how she wants to handle the second scenario